### PR TITLE
Cluster template

### PR DESCRIPTION
**Pull request by prachidamle**

### CODE REVIEW

File: apis/management.cattle.io/v3/cluster_template_types.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+}
+
+type ClusterTemplateSpec struct {
+	DisplayName         string `json:"displayName" norman:"required"`
+	Description         string `json:"description"`
+	Disabled            bool   `json:"disabled,omitempty"`
+	DefaultRevisionName string `json:"defaultRevisionName,omitempty" norman:"type=reference[clusterTemplateRevision]"`
+
+	Members       []Member `json:"members,omitempty"`
+	RevisionLimit int      `json:"revisionLimit,omitempty" norman:"default=10"`
```
Comment (alena1108) [inline apis/management.cattle.io/v3/cluster_template_types.go]: We might not need it anymore as create/delete of revisions is controlled via API

File: apis/management.cattle.io/v3/cluster_template_types.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+	// https://github.com/kubernetes/community/blob/master/contributors/devel/api-conventions.md#spec-and-status
+	Spec ClusterTemplateRevisionSpec `json:"spec"`
+}
+
+type ClusterTemplateRevisionSpec struct {
+	Disabled            bool   `json:"disabled,omitempty"`
+	IsDefault           bool   `json:"isDefault,omitempty"`
+	ClusterTemplateName string `json:"clusterTemplateName,omitempty" norman:"type=reference[clusterTemplate],required"`
+	// Kubernetes version to use
+	Version       string         `yaml:"kubernetes_version" json:"kubernetesVersion,omitempty"`
```
Comment (alena1108) [inline apis/management.cattle.io/v3/cluster_template_types.go]: Have we decided on a top level version, or will we use k8s version from clusterConfig. RancherKubernetesEngineConfig?

Comment (prachidamle) [reply apis/management.cattle.io/v3/cluster_template_types.go]: We can just use the clusterConfig.RancherKubernetesEngineConfig.Version. I will remove the extra parameter.

File: apis/management.cattle.io/v3/cluster_template_types.go (hunk 1 above)
Comment (prachidamle) [reply apis/management.cattle.io/v3/cluster_template_types.go]: Will remove this limit.

**Comment by alena1108** [comment]
LGTM

**Pull request merged**